### ISSUE TITLE
Fennel forms query issues

### ISSUE THREAD

**Issue by NoahTheDuke** [julienvincent/nvim-paredit#83]
It seems the fennel queries aren't quite correct, leading to odd behavior:

```fennel
{:foo 1
 |:bar 2
 :qux 3}
```
Given my cursor on the `:` of `:bar`, if I press `W`, my cursor will move to `:qux`, not `2`. And if I press `B`, it'll move to `:foo`, not `1`.

**Comment by julienvincent** [comment]
Interesting. I wonder how this worked before..

Looking at the grammar it looks like the table elements are wrapped in a 'table_pair' node. See:

```scm
(program ; [0, 0] - [3, 0]
  (table ; [0, 0] - [2, 8]
    item: (table_pair ; [0, 1] - [0, 7]
      key: (string ; [0, 1] - [0, 5]
        content: (string_content)) ; [0, 2] - [0, 5]
      value: (number)) ; [0, 6] - [0, 7]
    item: (table_pair ; [1, 1] - [1, 7]
      key: (string ; [1, 1] - [1, 5]
        content: (string_content)) ; [1, 2] - [1, 5]
      value: (number)) ; [1, 6] - [1, 7]
    item: (table_pair ; [2, 1] - [2, 7]
      key: (string ; [2, 1] - [2, 5]
        content: (string_content)) ; [2, 2] - [2, 5]
      value: (number)))) ; [2, 6] - [2, 7]
```

So if this worked on the previous paredit version it was purely by chance.

I will need to think on how to solve this. It might require introducing more capture groups.

**Comment by julienvincent** [comment]
Ah, as per https://github.com/julienvincent/nvim-paredit-fennel/pull/1 it looks like the fennel grammar changed earlier this year. So the old paredit would have been broken on the latest fennel grammar anyway.

**Comment by NoahTheDuke** [comment]
Yeah, I have the latest edition of the fennel grammar. I tried to write my own query according to the docs but wasn't able to get it. I wish the grammar was closer to how Clojure's grammar works. Ah well.

**Comment by julienvincent** [comment]
Yea you wouldn't be able to with how paredit currently works.

Paredit try to find the root node relative to the closest form, starting from the node at the cursor.

```
(program
  (form
   (outer_element
    |(inner_element))
   
   (sibling)))
```

In the above example the `outer_element` and `sibling` nodes are considered to be the 'elements' of the form.

The paredit grammars are only used to identify the nodes considered forms and the rest is inferred. So we would need to extend paredit to allow capturing the distinct 'elements' of a form in a way that is independent of the structure of the grammar.

This would mean supporting a query like:

```
(program
  (table
    (table_pair
      (_) @form.element)) @form)
```
